e2e/tests-dfx/build.bash

```diff
@@ -215,7 +215,7 @@ teardown() {
   jq '.canisters.e2e_project_backend.type="unknown_canister_type"' dfx.json | sponge dfx.json
   assert_command_fail dfx build
   # shellcheck disable=SC2016
-  assert_match 'unknown variant `unknown_canister_type`'
+  assert_match 'ExtensionNotInstalled("unknown_canister_type")' # TODO: concrete error type
```

Comment:
Open TODO

Reply:
relates to https://github.com/dfinity/sdk/pull/3222#discussion_r1376255304